src/common/config.cc

```diff
@@ -1227,6 +1227,47 @@ void md_config_t::diff(
   }
 }
 
+void md_config_t::diff_setting(const md_config_t *other,
+            std::map<std::string, std::pair<std::string, std::string>> *diff, 
+            std::set<std::string> *unknown, const std::string& ceph_setting, bool show_unchanged)
+{
+  Mutex::Locker l(lock);
+
+  bool wasItFound = false;
+  char local_buf[4096];
+  char other_buf[4096];
+  for (int i = 0; i < NUM_CONFIG_OPTIONS; i++) {
+    if (wasItFound) break;  //-- If we already found it, we are done.
+    config_option *opt = &config_optionsp[i];
+    std::string ceph_setting_name(opt->name);
+    if (ceph_setting == ceph_setting_name) {
+      memset(local_buf, 0, sizeof(local_buf));
+      memset(other_buf, 0, sizeof(other_buf));
+
+      char *other_val = other_buf;
+      int err = other->get_val(opt->name, &other_val, sizeof(other_buf));
+      if (err < 0) {
+        if (err == -ENOENT) {
+          unknown->insert(opt->name);
```

Comment:
@badone, but 'get a test' do you mean "Add a test for it in the test_config.cc"? 

Thanks.